🐛 BUG: vite plugin with top level await (wasm) dev build

### Which Cloudflare product(s) does this pertain to?

Vite Plugin

### What versions are you using?

4.0.0 [Wrangler], 0.1.11 [@cloudflare/vite-plugin], 22.9.0 [Node.js], 6.2.2 [vite], 1.4.6 [loro-crdt]

### What operating system and version are you using?

macOS 15

### Please provide a link to a minimal reproduction

https://github.com/npaton/cf-vite-loro-repro

### Describe the Bug

Using `@cloudflare/vite-plugin` and [`loro-crdt`](https://www.npmjs.com/package/loro-crdt) together does not work. Loro has a top level await to load its wasm. A dev build (`wrangler dev`) fails with the following error:

```
✘ [ERROR] Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)

    node_modules/.pnpm/loro-crdt@1.4.6/node_modules/loro-crdt/bundler/loro_wasm.js:10:18:
      10 │     const wkmod = await import("./loro_wasm_bg.wasm");
         ╵                   ~~~~~

node_modules/.pnpm/esbuild@0.25.1/node_modules/esbuild/lib/main.js:1477
  let error = new Error(text);
              ^

Error: Build failed with 1 error:
node_modules/.pnpm/loro-crdt@1.4.6/node_modules/loro-crdt/bundler/loro_wasm.js:10:18: ERROR: Top-level await is not available in the configured target environment ("chrome87", "edge88", "es2020", "firefox78", "safari14" + 2 overrides)
    at failureErrorWithLog (node_modules/.pnpm/esbuild@0.25.1/node_modules/esbuild/lib/main.js:1477:15)
    at node_modules/.pnpm/esbuild@0.25.1/node_modules/esbuild/lib/main.js:946:25
    at node_modules/.pnpm/esbuild@0.25.1/node_modules/esbuild/lib/main.js:1355:9
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5) {
  errors: [Getter/Setter],
  warnings: [Getter/Setter]
}
```

The `vite.config.ts` file:

```ts
import { cloudflare } from "@cloudflare/vite-plugin";
import react from "@vitejs/plugin-react";
import { defineConfig } from "vite";
import topLevelAwait from "vite-plugin-top-level-await";
import wasm from "vite-plugin-wasm";

export default defineConfig({
  plugins: [react(), cloudflare(), wasm(), topLevelAwait()],
});
```

A `preview` build (`tsc -b && vite build && vite preview`) works fine, it's only in dev (`vite`) that it fails.

Note that without the cloudflare vite plugin, this setup works (top level await and wasm plugins).

The intention was to use loro-crdt on both client and server. But even if I do not import loro-crdt on the client side, the build fails. loro-crdt works fine on the server side (if I stub the resolve on the client side or create a standalone workers project).

I cannot change the `build.target` (with the intention of circumventing the issue), which seems to be overridden by the cloudflare plugin.

### Please provide any relevant error logs

_No response_

--- a/.changeset/nasty-bars-sniff.md
+++ b/.changeset/nasty-bars-sniff.md
@@ -1,0 +1,5 @@
+---
+"@cloudflare/vite-plugin": patch
+---
+
+Set `target` in `optimizeDeps.esbuildOptions` to `es2022`. This fixes a bug where the target for prebundled dependencies did not match the build target.

--- a/packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
+++ b/packages/vite-plugin-cloudflare/src/cloudflare-environment.ts
@@ -119,6 +119,7 @@ const cloudflareBuiltInModules = [
 ];
 
 const defaultConditions = ["workerd", "module", "browser"];
+const target = "es2022";
 
 export function createCloudflareEnvironmentOptions(
 	workerConfig: WorkerConfig,
@@ -144,7 +145,7 @@ export function createCloudflareEnvironmentOptions(
 			createEnvironment(name, config) {
 				return new vite.BuildEnvironment(name, config);
 			},
-			target: "es2022",
+			target,
 			// We need to enable `emitAssets` in order to support additional modules defined by `rules`
 			emitAssets: true,
 			outDir: getOutputDirectory(userConfig, environmentName),
@@ -165,6 +166,7 @@ export function createCloudflareEnvironmentOptions(
 			exclude: [...cloudflareBuiltInModules],
 			esbuildOptions: {
 				platform: "neutral",
+				target,
 				conditions: [...defaultConditions, "development"],
 				resolveExtensions: [
 					".mjs",

